src/Test/Utilities/Desktop/CommonDiagnosticAnalyzers.cs

```diff
@@ -480,120 +480,5 @@ private void OnCompilation(CompilationAnalysisContext context)
-                context.EnableConcurrentExecution();
-            }
-
-            private void OnCompilationStart(CompilationStartAnalysisContext context)
-            {
-                Assert.True(context.Compilation.Options.ConcurrentBuild, "This analyzer is intended to be used only when concurrent build is enabled.");
-
-                var pendingSymbols = new ConcurrentSet<INamedTypeSymbol>();
-                foreach (var type in context.Compilation.GlobalNamespace.GetTypeMembers())
-                {
-                    if (_symbolNames.Contains(type.Name))
-                    {
-                        pendingSymbols.Add(type);
-                    }
-                }
-
-                context.RegisterSymbolAction(symbolContext =>
-                {
-                    if (!pendingSymbols.Remove((INamedTypeSymbol)symbolContext.Symbol))
-                    {
-                        return;
-                    }
-
-                    var myToken = Interlocked.Increment(ref _token);
-                    if (myToken == 1)
-                    {
-                        // Wait for all symbol callbacks to execute.
-                        // This analyzer will deadlock if the driver doesn't attempt concurrent callbacks.
-                        while (pendingSymbols.Any())
-                        {
-                            Thread.Sleep(10);
-                        }
-                    }
-
-                    // ok, now report diagnostic on the symbol.
-                    var diagnostic = Diagnostic.Create(Descriptor, symbolContext.Symbol.Locations[0], symbolContext.Symbol.Name);
-                    symbolContext.ReportDiagnostic(diagnostic);
-                }, SymbolKind.NamedType);
-            }
-        }
-
-        /// <summary>
-        /// This analyzer will report diagnostics only if it receives any concurrent action callbacks, which would be a
-        /// bug in the analyzer driver as this analyzer doesn't invoke <see cref="AnalysisContext.RegisterConcurrentExecution"/>.
-        /// </summary>
-        [DiagnosticAnalyzer(LanguageNames.CSharp, LanguageNames.VisualBasic)]
-        public class NonConcurrentAnalyzer : DiagnosticAnalyzer
-        {
-            public static readonly DiagnosticDescriptor Descriptor = new DiagnosticDescriptor(
-                "NonConcurrentAnalyzerId",
-                "Title",
-                "Analyzer driver made concurrent action callbacks, when analyzer didn't register for concurrent execution",
-                "Category",
-                DiagnosticSeverity.Warning,
-                true);
-            private const int registeredActionCounts = 1000;
-
-            public override ImmutableArray<DiagnosticDescriptor> SupportedDiagnostics => ImmutableArray.Create(Descriptor);
-            public override void Initialize(AnalysisContext context)
-            {
-                SemaphoreSlim gate = new SemaphoreSlim(initialCount: registeredActionCounts);
```

Comment:
Ah, the new test I added also used the SemaphoreSlim utilities. Can you retains these tests and also the move of SemaphoreSlim utilities to CodeAnalysis? It will have test only consumers for now, but I hope to restore the original PR with the fix soon.